### PR TITLE
Preselected rights

### PR DESCRIPTION
**Pull request by Mussuranae**
Expansion ok, form ok, need to connect to bucket for the add to selection.

### CODE REVIEW

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html [hunk 1]
```diff
@@ -92,18 +92,150 @@
...
+              </article>
+            </mat-expansion-panel>
 
-  <mat-divider></mat-divider>
+            <!-- VERSIONS -->
+            <mat-expansion-panel>
+              <mat-expansion-panel-header>
+                <mat-panel-title>
+                  Versions
+                  <span>&nbsp;(Non Mandatory)</span>
```
Comment (RemcoSimonides) [inline apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html:139]: Why is it in a span? It is shown in the same font style as Versions, right?

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html [hunk 2]
```diff
@@ -92,18 +92,150 @@
...
+                <article formGroupName="languages" fxLayout="column" fxLayoutGap="24px">
+                  <ng-container *ngFor="let language of form.controls.languages.controls | keyvalue">
+                    <div fxLayout="column" fxLayoutGap="8px">
+                      <p class="uppercase">{{ language.key | toLabel: 'languages' }}</p>
+                      <div [formGroupName]="language.key" fxLayout="row" fxLayoutAlign="start center" fxLayoutGap="16px">
+                        <div fxLayout="row" fxLayoutGap="16px">
+                          <mat-checkbox formControlName="subtitle" color="primary">Subtitles</mat-checkbox>
+                          <mat-checkbox formControlName="dubbed" color="primary">Dubbings</mat-checkbox>
+                          <mat-checkbox formControlName="caption" color="primary">Closed Captions</mat-checkbox>
+                        </div>
```
Comment (RemcoSimonides) [inline apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html:157]: Is it needed to have a row in a row? Without the inner row the mat-checkboxes will also be aligned horizontally, right? with gap of 16px

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html [hunk 3]
```diff
@@ -92,18 +92,150 @@
...
+                    </button>
+                  </div>
+                </div>
+                <div fxLayout fxLayoutAlign="start baseline" fxLayoutGap="16px" *ngIf="showButtons">
+                  <mat-divider inset></mat-divider>
+                  <span class="mat-caption text-primary">Add a Version</span>
+                  <button mat-mini-fab color="primary" (click)="showForm()">
+                    <mat-icon svgIcon="add"></mat-icon>
+                  </button>
+                </div>
```
Comment (RemcoSimonides) [inline apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html:189]: Let's change this to a
```
<ng-container *ngIf="showButtons else noButtons">
</ng-container>
<ng-template #noButtons>
</ng-template>
```

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html [hunk 4]
```diff
@@ -92,18 +92,150 @@
...
 
-</mat-card>
\ No newline at end of file
+                <!-- ADD LANGUAGE -->
+                <div *ngIf="!showButtons">
+                  <input-autocomplete test-id="languages" scope="languages" [control]="languageCtrl" withoutValues="['all']">
+                    <mat-label>Languages</mat-label>
+                  </input-autocomplete>
+                  <div fxLayout="row" fxLayoutAlign="space-between start">
+                    <button mat-stroked-button [disabled]="languageCtrl.invalid" (click)="addLanguage()"
```
Comment (RemcoSimonides) [inline apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html:176]: Color attribute still needs to be defined I think?

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts [hunk 5]
```diff
@@ -38,12 +50,19 @@ export class MarketplaceMovieAvailsComponent implements OnInit {
...
+    private bucketQuery: BucketQuery,
+    private bucketService: BucketService
   ) { }
 
   public async ngOnInit() {
     this.org$ = this.orgService.valueChanges(this.movieQuery.getActive().orgIds[0]);
+    this.bucket$ = this.bucketQuery.selectActive();
+    this.form.valueChanges.subscribe(value => {
+      console.log(value)
+    })
```
Comment (RemcoSimonides) [inline apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts:62]: Remove

You can do `this.form.valueChanges.subscribe(console.log)` btw :)

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts [hunk 6]
```diff
@@ -68,4 +87,21 @@ export class MarketplaceMovieAvailsComponent implements OnInit {
...
+    this.form.get('languages').addControl(this.languageCtrl.value, new VersionSpecificationForm(spec));
+    this.languageCtrl.reset();
+    this.showButtons = true;
+  }
+
+  deleteLanguage(language: Language) {
+    this.form.controls.languages.removeControl(language);
+  }
+
+  showForm() {
```
Comment (RemcoSimonides) [inline apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts:101]: This simple toggle is often done directly in the html
`(click)="this.showButtons = !this.showButtons"`

**Review by RemcoSimonides — APPROVED** [review]

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html (hunk 1 above)
Comment (Mussuranae) [reply apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html:139]: No it looks like the `Version` is in bold and the `mandatory` not. So span helps to differentiate the two typo.

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html (hunk 2 above)
Comment (Mussuranae) [reply apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html:157]: Yes you're right. I removed it.

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html (hunk 4 above)
Comment (Mussuranae) [reply apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.html:176]: Could be yes, but I have no button in fact on the mockup. I guess I have to see that with the design team.

**Pull request merged**